Projects/Shapes/Shapes/Shapes/ShapeBuilder.cpp

```diff
@@ -0,0 +1,369 @@
+		std::cout << "Area: " << Array[i]->GetArea() << std::endl;
+		std::cout << "Perimeter: " << Array[i]->GetPerimeter() << std::endl;
+	}
+	RequestEnterFromInput();
+}
+
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::DeleteArray(FShape** Array, const int NumberOfShapes) const
+{
+	for (int i = 0; i < NumberOfShapes; i++)
+	{
+		delete Array[i];
+		Array[i] = nullptr;
+	}
+	delete Array;
+	Array = nullptr;
+}
+
+
+//-------------------------------------------------------------------------------------------
+//										 STATIC
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::AskForShapesStatic()
+{
+	system("CLS");
+	std::cout << "You will now be asked to provide information for 5 Shapes." << std::endl;
+
+	//Fill Array
+	TStaticArray<FShape*, 5> Shapes = FillStaticArray();
+
+	//Print Array
+	PrintArray(Shapes);
+	RequestEnterFromInput();
+
+	//Delete Array
+	DeleteArray(Shapes);
+
+	ActiveCommand = EMenuCommand::MainMenu;
+}
+
+//-------------------------------------------------------------------------------------------
+TStaticArray<FShape*, 5> FShapeBuilder::FillStaticArray() const
+{
+	TStaticArray<FShape*, 5> Shapes;
+
+	RequestEnterFromInput();
+
+	for (int i = 0; i < 5; i++)
+	{
+		system("CLS");
+		Shapes[i] = CreateShape();
+	}
+
+	return Shapes;
+}
+
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::PrintArray(TStaticArray<FShape*, 5>& Array) const
+{
+	system("CLS");
+
+	for (int i = 0; i < Array.GetSize(); i++)
+	{
+		std::cout << "Shape " << i+1 << std::endl;
+		std::cout << "Area: " << Array[i]->GetArea() << std::endl;
+		std::cout << "Perimeter: " << Array[i]->GetPerimeter() << std::endl;
+	}
+	RequestEnterFromInput();
+}
+
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::DeleteArray(TStaticArray<FShape*, 5>& Array) const
+{
+	for (int i = 0; i < Array.GetSize(); i++)
+	{
+		delete Array[i];
+		Array[i] = nullptr;
+	}
+}
+
+
+
+//-------------------------------------------------------------------------------------------
+//										 DYNAMIC
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::AskForShapesDynamic()
+{
+	system("CLS");
+	std::cout << "We're gonna add your shapes to a dynamic array!" << std::endl;
+	RequestEnterFromInput();
+
+	//Fill Array
+	TDynamicArray<FShape*> Shapes = FillDynamicArray();
+
+	//Print Array
+	PrintArray(Shapes);
+	RequestEnterFromInput();
+
+	//Delete
+	DeleteArray(Shapes);
+
+	ActiveCommand = EMenuCommand::MainMenu;
+}
+
+//-------------------------------------------------------------------------------------------
+TDynamicArray<FShape*> FShapeBuilder::FillDynamicArray() const
+{
+	TDynamicArray<FShape*> Shapes;
+
+
+	int Continue = 1;
```

Comment:
An int called continue doesn't tell me anything. Maybe this could be a bool? And so you could assign to the bool inside the loop.